tests/cypress/e2e/unit_tests/rbac_fleet.spec.ts

```diff
@@ -26,45 +26,75 @@ describe('Test Fleet access with RBAC with custom roles', { tags: '@rbac' }, ()
-    })
+  const roleTypeTemplate = "Global"
+
+
+  const rbacTestData: testData[] = [
+    { 
+      qase_id: 5,
+      test_explanation: 'NOT "Clusters" NOR "Clusters Groups"',
+      defaultBaseRole: "User-Base",
+    },
+    {
+      qase_id: 42,
+      test_explanation: 'NOT "Cluster Registration Tokens" "BundleNamespaceMappings" and "GitRepoRestrictions"',
+      defaultBaseRole: "Standard User",
+    }
+  ]
+
+  rbacTestData.forEach(
+    ({qase_id, test_explanation, defaultBaseRole}) => {
+      qase(`${qase_id}`,
+      it(`Test ${defaultBaseRole} role user with custom role to "fleetworkspaces", "gitrepos" and "bundles" and  ALL verbs access CAN access "Workspaces", "Bundles" and "Git Repos" but ${test_explanation}`, { tags: `@fleet-${qase_id}` }, () => {
+        const customRoleName = "fleetworkspaces-bundles-gitrepos-all-verbs-role"
+
+        // Create User "User-Base"
+        cypressLib.burgerMenuToggle();
+        cypressLib.createUser(baseUser, uiPassword, defaultBaseRole, true);
+
+        cy.createRoleTemplate({
+          roleType: roleTypeTemplate,
+          roleName: customRoleName,
+          rules: [
+            { resource: "fleetworkspaces", verbs: ["create", "delete", "get", "list", "patch", "update", "watch"]},
+            { resource: "gitrepos", verbs: ["create", "delete", "get", "list", "patch", "update", "watch"]},
+            { resource: "bundles", verbs: ["create", "delete", "get", "list", "patch", "update", "watch"]},
+          ]
+        });
+
+        // Assign role to the created user
+        cy.assignRoleToUser(baseUser, customRoleName);
+
+        // Logout as admin and login as other user
+        cypressLib.logout();
+        cy.login(baseUser, uiPassword);
+
+        // Ensuring the user IS able to "go to Continuous Delivery", "list" and "delete" workspaces.
+        cy.accesMenuSelection('Continuous Delivery', 'Git Repos');
+        cy.wait(500);
+        cy.accesMenuSelection('Continuous Delivery', 'Advanced', 'Workspaces');
+        cy.accesMenuSelection('Continuous Delivery', 'Advanced', 'Bundles');
+
+        // Ensuring user cannot access Clusters nor Clusters Groups
+        cy.accesMenuSelection('Continuous Delivery');
+        cy.contains('Clusters').should('not.exist');
+        cy.contains('Clusters Groups').should('not.exist');
+
+        // FLEET-42 test cases extra check.
+        if (qase_id === 42) {
+          cy.accesMenuSelection('Continuous Delivery', 'Advanced');
+          cy.contains('Cluster Registration Tokens').should('not.exist');
+          cy.contains('GitRepoRestrictions').should('not.exist');
+          cy.contains('BundleNamespaceMappings').should('not.exist');
+        }
+        // Logout other user and login as admin user to perform user and role cleanup
+        cypressLib.logout();
+        cy.login();
+        cy.deleteUser(baseUser);
+        cy.deleteRole(customRoleName, roleTypeTemplate.toUpperCase());
+      })
+    )
+    }
```

Comment:
Sorry, but this is very difficult to read in this kind of tests with lots of small configs enabled / disabled etc.

I see the idea behind trying to reuse code, but it makes complicated going back and forth to get to know test title, what is expected and so on and it can be prone to errors.

If we want to reutilize code let's pls create an isolated Describe using as boilerplate user and custom role creation and later 2 it with individual descriptions solely with the login with each specific user

Aside from this, I see that a method to delete `baseUser` `cy.deleteUser(baseUser)` is present.  Given that one user is `standard` and the other is `base`, will the `standard` one be deleted also?

Reply:
Okay, I got your point and it is true as well. I will create a separate `describe` for QASE-5 and QASE-42 by re-using common code.

Also, yes, `cy.deleteUser(baseUser)` will delete both `Standard User` and `User Base` as I passed their values from the `for-loop`.

Reply:
Thanks for the refactoring.
If the delete function will delete 2 users given 1 was only passed in parametrization then I find it kind of confusing. 
It would make sense either not to pass any specific parameter or to specifically call the 2 of them in my view.